sql/opt_split.cc

```diff
@@ -367,8 +368,8 @@ bool JOIN::check_for_splittable_materialized()
   ORDER *partition_list= 0;
   st_select_lex_unit *unit= select_lex->master_unit();
   TABLE_LIST *derived= unit->derived;
-  if (!(optimizer_flag(thd, OPTIMIZER_SWITCH_SPLIT_MATERIALIZED)) ||  // !(1)
-      !(derived && derived->is_materialized_derived()) ||             // !(2)
+
+  if (!(derived && derived->is_materialized_derived()) ||             // !(2)
```

Comment:
Numeration now starts from 2, I'd suggest adjusting it here and in the function description above. The new condition `if (!is_split_materialized_allowed)` can also be marked with some number and reflected in the description